File: MyLeetCodeAnswers/java/ReverseLinkedList.java
/**
 * Source   : https://leetcode.com/problems/reverse-linked-list/
 * Author   : Wang Meng
 * Date     : 2016-03-21
 * <p>
 * Reverse a singly linked list.
 */
public class ReverseLinkedList {
    public ListNode reverseList(ListNode head) {
        if (head == null) return null;
        ListNode pre = null;
        ListNode cur = head;
        ListNode next;
        while (cur != null) {
            next = cur.next;
            cur.next = pre;
            pre = cur;
            cur = next;
        }
        return pre;
    }
}

File: MyLeetCodeAnswers/python/ReverseLinkedList.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
# Reverse a singly linked list.

# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

from leetcommon import ListNode

class Solution(object):
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        pre = None
        cur = head
        while cur != None:
            nextp = cur.next
            cur.next = pre
            pre = cur
            cur = nextp
        return pre

head = None
#head = ListNode(1)
#head.next = ListNode(2)
#head.next.next = ListNode(3)
s = Solution()
print s.reverseList(head)
